src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -782,6 +784,38 @@ namespace AirLoopHVACDOAS {
 
                 thisDOAS.m_AirLoopDOASNum = AirLoopDOASNum - 1;
                 state.dataAirLoopHVACDOAS->airloopDOAS.push_back(thisDOAS);
+
+                // ensure the inlet node should be an outdoor air node
+                bool OAnodeFound = false;
+                for (int OutsideAirNodeNum = 1; OutsideAirNodeNum <= state.dataOutAirNodeMgr->NumOutsideAirNodes; ++OutsideAirNodeNum) {
+                    if (thisDOAS.m_InletNodeNum == state.dataOutAirNodeMgr->OutsideAirNodeList(OutsideAirNodeNum)) {
+                        OAnodeFound = true;
+                        break;
+                    }
+                }
+                if (!OAnodeFound) {
+                    ShowSevereError(state,
+                                    format("Inlet node ({}) is not one of OutdoorAir:Node in {} = {}",
+                                           state.dataLoopNodes->NodeID(thisDOAS.m_InletNodeNum),
+                                           CurrentModuleObject,
+                                           thisDOAS.Name));
+                    errorsFound = true;
+                }
+                // Ensure the outlet node is the splitter inlet node 
+                bool OutletnodeFound = false;
+                if (thisDOAS.m_OutletNodeNum == thisDOAS.m_CompPointerAirLoopSplitter->InletNodeNum) {
+                    OutletnodeFound = true;
+                }
+                if (!OutletnodeFound) {
+                    ShowSevereError(
+                        state,
+                        format("Outlet node is not the inlet node of AirLoopHVAC:Splitter in {} = {}", CurrentModuleObject, thisDOAS.Name));
+                    ShowContinueError(state,
+                                      format("The outlet node name is {}, and the inlet node name of AirLoopHVAC:Splitter is {}",
+                                             state.dataLoopNodes->NodeID(thisDOAS.m_OutletNodeNum),
+                                             state.dataLoopNodes->NodeID(thisDOAS.m_CompPointerAirLoopSplitter->InletNodeNum)));
+                    errorsFound = true;
+                }
```

Comment:
Issue severe errors to ensure the inlet node of the first component is one of outdoor air nodes and the outlet node of the last component is the inlet node of AirLoopHVAC:Splitter.